e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/utilisationReportPage.js

```diff
@@ -28,7 +28,16 @@ const utilisationReportPage = {
         cy.get(`[type="checkbox"][id="feeRecordIds-${feeRecordIds.join(',')}-reportedPaymentsCurrency-${paymentCurrency}-status-${status}"]`),
       status: (feeRecordId) => cy.get(`[data-cy="premium-payments-table-row--feeRecordId-${feeRecordId}-status"]`),
       // NOTE: Using the ^= selector to match all table rows starting with the given data-cy value (regardless of the actual fee record id value).
-      facilityIdByRowIndex: (rowIndex) => cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('th').first(),
+      rows: () => cy.get(`tr[data-cy^="premium-payments-table-row--feeRecordId-"]`),
+      facilityIdByRowIndex: (rowIndex) =>
+        cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('[data-cy="facility-id"]').first(),
+      exporterIdByRowIndex: (rowIndex) => cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('[data-cy="exporter"]').first(),
+      reportedFeesByRowIndex: (rowIndex) =>
+        cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('[data-cy="reported-fees"]').first(),
+      reportedPaymentsByRowIndex: (rowIndex) =>
+        cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('[data-cy="reported-payments"]').first(),
+      totalReportedPaymentsByRowIndex: (rowIndex) =>
+        cy.get('tr[data-cy^="premium-payments-table-row--feeRecordId-"]').eq(rowIndex).find('[data-cy="total-reported-payments"]').first(),
```

Comment:
Can we have a `rowIndex` object instead?

```js
rowIndex: {
  facilityId: () => '...',
  exporterId: () => '...',
  // etc
}
```

Reply:
We have to pass the row index in as a parameter and I dislike:`rowIndex.facilityId(rowIndex)` since the parameter is removed from the statement that claims to care about it. 

I guess I could try and define it as:
```
rowIndex: (index) => ({
  facilityId: () => '...some selector with index in it...',
  exporterId: () => '...some selector with index in it...',
  // etc
}),
```
So then would be called like `rowIndex().facilityId()` but this feels like a different pattern to everything I have seen until now so keen to know what you think before I try it?

Reply:
I think the current selectors are too long and not very readable, e.g `totalReportedPaymentsByRowIndex` or `xyzByRowIndex`

There's only 1x existing selector that has `xyzByRowIndex` (facilityIdByRowIndex)

I think a `rowIndex` function like you mentioned is a lot cleaner. What do you think? e.g

```js
rowIndex: (index) => ({
  facilityId: () => cy.get(`[data-cy="row-${index}-facility-id"]`),
  exporterId: () => cy.get(`[data-cy="row-${index}-exporter-id"]`),
})
```

Reply:
I have switched to this pattern but combined with the double selector trick you shared to use the exporter directly, this is now greatly simplified! Thanks for the suggestions, especially the double data-cy, makes this so much nicer!

Reply:
Applied the same pattern to the utilisation tab tests for consistency whilst I was at it as well